Slow down, animations, teleport, and small stuff

b-irsigler opened the pull request:
closes #57 
closes #59 
closes #66

b-irsigler committed:
Implement slow down

Christine is slowed down by 5% of the default speed per beech she is carrying. Both motion and animation speed are updated. When returning the beeches at the castle, the default speed is restored. Hotkeys for adjusting motion and animation speed are still functional.

b-irsigler committed:
Adjust motion speed of NPCs

Komtur and Der Gruene go at 85% and Die Spinne goes at 75% of Christine's default speed. This means, if Christine carries less than three beeches, she can run away from the Komtur as well as from Die Spinne carrying less than five beeches.

b-irsigler committed:
Add attack animation for Spinne

Had to introduce a new timer. The behavior should be reviewed at some point.

b-irsigler committed:
Add Komtur attack animation

Attack signal is emitted after the attack timer has timed out. Perhaps this is not intended.

b-irsigler committed:
Fixed margin of castle indicator

Scaling the windows still leads to the displacement of the indicator. This is more difficult to fix.

b-irsigler committed:
Implemented teleport for Spinne and Der Gruene

with probability Chrisitine will encounter these NPCs. There is code dupblication, should be refactored at some point.

d-schn reviewed (approved):
I would merge this PR now, however some issues with the state machines remain:
1. Attack state persisting for more than one cycle is not good if attack state triggers sounds, signals, etc, which I think it should
2. Animation during attack/cooldown seems to be repeating? Maybe because of 1.
3. Entering attack state should be tied to current state = chase, otherwise reattacking can occure before cooldown ends.

d-schn committed:
State machine changes

Attack state only lasts fro one cycle, so any attack related functions only happen once
If attack area is exited, no state changes are triggered
If chase area is exited, state only changes if it was chase before
